src/extension.ts

```diff
@@ -50,6 +51,9 @@ export async function activate(context: vscode.ExtensionContext): Promise<IExten
             const connectionSucceeded = await connectionPromise;
             if (!connectionSucceeded) {
                 throw new Error(`Connection for ${JSON.stringify(connectionInfo)} failed`);
+            } else if (saveConnection) {
+                let profile: IConnectionProfile = <IConnectionProfile>connectionInfo;
```

Comment:
const instead of let

Reply:
right now we assume they are all optional but  I think the correct to handle it is the API accept IConnectionInfo or IConnectionProfile so that if anyone wants to save, have to pass IconnectionProfile